File: bindings/go/libc_from_ldd.go
package memless

import "bytes"

func libcFromLdd(data []byte, err error) string {
	if err != nil {
		return ""
	}
	if bytes.Contains(data, []byte("musl")) {
		return "musl"
	}
	if bytes.Contains(data, []byte("GNU C Library")) {
		return "glibc"
	}
	return ""
}

File: bindings/node/src/libc-from-ldd.js
'use strict';

function libcFromLdd(text) {
  if (!text) {
    return null;
  }
  if (text.includes('musl')) {
    return 'musl';
  }
  return text.includes('GNU C Library') ? 'glibc' : null;
}

module.exports = libcFromLdd;
